source/configuration.txt

```diff
@@ -11,53 +11,139 @@ Configuration Options
    :class: singlecol
 
 Various configuration options are available for the MongoDB Spark
-Connector.
+Connector. To learn more about the options you can set, see 
+:ref:`spark-write-conf` and :ref:`spark-read-conf`.
 
 Specify Configuration
 ---------------------
 
-Via ``SparkConf``
-~~~~~~~~~~~~~~~~~
+.. _spark-conf:
 
-You can specify these options via ``SparkConf`` using the ``--conf``
-setting or the ``$SPARK_HOME/conf/spark-default.conf`` file, and
-MongoDB Spark Connector will use the settings in ``SparkConf`` as the
+Using ``SparkConf``
+~~~~~~~~~~~~~~~~~~~
+
+You can specify configuration options with ``SparkConf`` using any of 
+the following approaches:
+
+.. tabs-selector:: drivers
+
+.. tabs-drivers::
+
+   tabs:
+     - id: java-sync
+       content: |
+
+         - The ``SparkConf`` constructor in your application. To learn more, see the `Java SparkConf documentation <https://spark.apache.org/docs/latest/api/java/index.html?org/apache/spark/SparkConf.html>`__.
+
+     - id: python
+       content: |
+
+         - The ``SparkConf`` constructor in your application. To learn more, see the `Python SparkConf documentation <https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.SparkConf.html>`__.
+
+     - id: scala
+       content: |
+
+         - The ``SparkConf`` constructor in your application. To learn more, see the `Scala SparkConf documentation <https://spark.apache.org/docs/latest/api/scala/org/apache/spark/SparkConf.html>`__.
+
+- The ``--conf`` flag at runtime. To learn more, see 
+  `Dynamically Loading Spark Properties <https://spark.apache.org/docs/latest/configuration.html#dynamically-loading-spark-properties>`__ in 
+  the Spark documentation.
+
+- The ``$SPARK_HOME/conf/spark-default.conf`` file.
+
+The MongoDB Spark Connector will use the settings in ``SparkConf`` as 
 defaults.
 
 .. important::
 
-   When setting configurations via ``SparkConf``, you must prefix the
-   configuration options. Refer to the configuration sections for the
-   specific prefix.
+   When setting configurations with ``SparkConf``, you must prefix the
+   configuration options. Refer to :ref:`spark-write-conf` and 
+   :ref:`spark-read-conf` for the specific prefixes.
 
-Via ``ReadConfig`` and ``WriteConfig``
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _read-write-config:
+
+Using ``ReadConfig`` and ``WriteConfig``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Various methods in the MongoDB Connector API accept an optional
```

Comment:
I think we can remove this paragraph. 

As the only API is the Spark api in v10 users are much less likely to use ReadConfig / WriteConfig objects/

Reply:
Do you think we should remove all references to Read and WriteConfig objects, or just not highlight them here? 

You can always use an options map instead, right? I can replace existing examples with equivalent options maps.

```
val df3 = sparkSession.loadFromMongoDB(ReadConfig(
     Map("uri" -> "mongodb://example.com/database.collection")
     )
   ) // ReadConfig used for configuration
```

Reply:
Yes definitely - can do in a different ticket. There are no implicit helpers eg: `sparkSession.loadFromMongoDB` because they were JVM only and not part of the official API.